proposals/p4246.md

```diff
@@ -0,0 +1,112 @@
+# Getting commit access
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/4246)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Longer idle times](#longer-idle-times)
+
+<!-- tocstop -->
+
+## Abstract
+
+Establish a process for getting commit access. We will:
+
+-   Grant access based on a developer's commit history.
+    -   Someone with commit access should nominate, and a contributor may ask.
+    -   A lead will approve nominations. Only one lead is needed.
+-   Remove commit access once someone is idle for 6 months.
```

Comment:
Added to background.

Reply:
Thanks!

You've addressed this to my satisfaction, but since you asked, here are a couple of examples of situations that I'd like to avoid:

- Someone behaves badly on github, let's say by merging PRs that it's clear should not have been merged, and we choose to remove their commit access, and they point at this proposal and say "you're not following your own documented rules" / "you're mistreating me".
- Someone looks at this proposal and decides they don't want to participate in our community because they see no remedy for people who misuse their commit access.

I think just something minimal -- like you've added -- that clarifies that these are not the *only* rules governing who gets to have commit access is sufficient.